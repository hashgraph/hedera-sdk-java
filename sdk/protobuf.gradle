dependencies {
    implementation 'io.grpc:grpc-protobuf:1.50.2'

    // Protobuf Lite is used to maintain easy compatibility with Android
    // https://github.com/protocolbuffers/protobuf/blob/master/java/lite.md
    api "com.google.protobuf:protobuf-java:3.21.8"
}

protobuf {
    generateProtoTasks {
        all()*.plugins {
            grpc {}
        }
    }
}
