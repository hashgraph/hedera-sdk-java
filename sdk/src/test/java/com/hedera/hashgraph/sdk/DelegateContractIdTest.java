package com.hedera.hashgraph.sdk;

import com.google.protobuf.InvalidProtocolBufferException;
import io.github.jsonSnapshot.SnapshotMatcher;
import org.bouncycastle.util.encoders.Hex;
import org.junit.AfterClass;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class DelegateContractIdTest {
    @BeforeAll
    public static void beforeAll() {
        SnapshotMatcher.start();
    }

    @AfterClass
    public static void afterAll() {
        SnapshotMatcher.validateSnapshots();
    }

    @Test
    void fromString() {
        SnapshotMatcher.expect(DelegateContractId.fromString("0.0.5005").toString()).toMatchSnapshot();
    }

    @Test
    void fromSolidityAddress() {
        SnapshotMatcher.expect(DelegateContractId.fromSolidityAddress("000000000000000000000000000000000000138D").toString()).toMatchSnapshot();
    }

    @Test
    void fromSolidityAddressWith0x() {
        SnapshotMatcher.expect(DelegateContractId.fromSolidityAddress("0x000000000000000000000000000000000000138D").toString()).toMatchSnapshot();
    }

    @Test
    void toBytes() throws InvalidProtocolBufferException {
        SnapshotMatcher.expect(Hex.toHexString(new DelegateContractId(5005).toBytes())).toMatchSnapshot();
    }

    @Test
    void fromBytes() throws InvalidProtocolBufferException {
        SnapshotMatcher.expect(DelegateContractId.fromBytes(new DelegateContractId(5005).toBytes()).toString()).toMatchSnapshot();
    }

    @Test
    void toSolidityAddress() {
        SnapshotMatcher.expect(new DelegateContractId(5005).toSolidityAddress()).toMatchSnapshot();
    }
}
