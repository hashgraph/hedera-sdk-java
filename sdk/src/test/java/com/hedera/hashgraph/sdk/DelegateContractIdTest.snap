com.hiero.sdk.DelegateContractIdTest.fromBytes=[
  "0.0.5005"
]


com.hiero.sdk.DelegateContractIdTest.fromSolidityAddress=[
  "0.0.5005"
]


com.hiero.sdk.DelegateContractIdTest.fromSolidityAddressWith0x=[
  "0.0.5005"
]


com.hiero.sdk.DelegateContractIdTest.fromString=[
  "0.0.5005"
]


com.hiero.sdk.DelegateContractIdTest.toBytes=[
  "188d27"
]


com.hiero.sdk.DelegateContractIdTest.toSolidityAddress=[
  "000000000000000000000000000000000000138d"
]
